app/controllers/admin/payout_reports_controller.rb

```diff
@@ -56,18 +56,26 @@ def upload_settlement_report
 
     EyeshadeClient.publishers.create_settlement(body: json)
 
+    not_found = []
+
     json.each do |entry|
       next unless entry["type"] == MANUAL && entry["owner"] && entry["amount"]
 
-      publisher_id = entry["owner"].sub(Publisher::OWNER_PREFIX, "")
-
-      invoice = Invoice.where(
-        publisher_id: publisher_id,
-        finalized_amount: entry["amount"],
-        status: Invoice::IN_PROGRESS
-      ).order(:created_at).first
-
-      next unless invoice.present?
+      begin
+        invoice = Invoice.find(entry.dig('documentId'))
+      rescue ActiveRecord::RecordNotFound
+        publisher_id = entry["owner"].sub(Publisher::OWNER_PREFIX, "")
+        invoice = Invoice.where(
+          publisher_id: publisher_id,
+          finalized_amount: (entry["probi"].to_i / 1E18).round(2),
```

Comment:
Would it be reasonable to eliminate this part of the `where` clause? It looks like it might return 0 records when in fact there is a record.

Reply:
If we take it out and there's multiple invoices which are being settled as part of the process this might mark the wrong one as paid